yb-voyager/cmd/importData.go

```diff
@@ -684,6 +699,60 @@ func importData(importFileTasks []*ImportFileTask) {
 
 }
 
+func importTasksViaTaskPicker(pendingTasks []*ImportFileTask, state *ImportDataState, progressReporter *ImportDataProgressReporter, poolSize int) error {
+	// The code can produce `poolSize` number of batches at a time. But, it can consume only
+	// `parallelism` number of batches at a time.
+	batchImportPool = pool.New().WithMaxGoroutines(poolSize)
+	log.Infof("created batch import pool of size: %d", poolSize)
+
+	taskPicker, err := NewSequentialTaskPicker(pendingTasks, state)
+	if err != nil {
+		return fmt.Errorf("create task picker: %w", err)
+	}
+	taskImporters := map[int]*FileTaskImporter{}
```

Comment:
what is this taskImporters map is for?
Are we reusing the taskImporter?
Once the task(file) is done, use of that taskImporter is also done?

Reply:
Also add a comment here

Reply:
yes, we are re-using the taskimporter. In every loop, we get the taskImporter and simply call `taskImporter.SubmitNextBatch()`

Reply:
I think the name is causing confusion :) 
It's more of a `ImportFileTaskBatchProducer` rather than a `FileTaskImporter`. It has a `SubmitBatch` function that lets us produce and submit a batch to the worker pool. 
So, we re-use the same instance of the struct untill all batches have been submitted.